File: data/gcc_versions_comparison/src/program/src/func29.c
#include "program.h" 
array_t* func29(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   unsigned int loop5 = 0;
   unsigned int loopLimit5 = (rand()%loopsFactor)/4 + 1;
   for(; loop5 < loopLimit5; loop5++) {
      if(PATH0 & 1) {
         array_t* array8;
         if (pCounter > 0) {
            array8 = vars->data[--pCounter];
            array8->refC++;
            DEBUG_COPY(array8->id);
         } else {
            array8 = (array_t*)malloc(sizeof(array_t));
            array8->size = 802;
            array8->refC = 1;
            array8->id = 8;
            array8->data = (unsigned int*)malloc(array8->size*sizeof(unsigned int));
            memset(array8->data, 0, array8->size*sizeof(unsigned int));
            DEBUG_NEW(array8->id);
         }
         unsigned int loop6 = 0;
         unsigned int loopLimit6 = (rand()%loopsFactor)/5 + 1;
         for(; loop6 < loopLimit6; loop6++) {
            for (int i = 0; i < array8->size; i++) {
               array8->data[i]--;
            }
            for (int i = 0; i < array8->size; i++) {
               if (array8->data[i] == 69) { 
                  return array8;
               }
            }
         }
         array8->refC--;
         if(array8->refC == 0) {
            free(array8->data);
            free(array8);
            DEBUG_FREE(array8->id);
         }
      }
      else {
         array_t_param params0;
         params0.size = 0;
         params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
         array_t* array9 = func40(&params0, loopsFactor);
         DEBUG_RETURN(array9->id);
         free(params0.data);
         array9->refC--;
         if(array9->refC == 0) {
            free(array9->data);
            free(array9);
            DEBUG_FREE(array9->id);
         }
      }
      array_t* array11;
      if (pCounter > 0) {
         array11 = vars->data[--pCounter];
         array11->refC++;
         DEBUG_COPY(array11->id);
      } else {
         array11 = (array_t*)malloc(sizeof(array_t));
         array11->size = 784;
         array11->refC = 1;
         array11->id = 11;
         array11->data = (unsigned int*)malloc(array11->size*sizeof(unsigned int));
         memset(array11->data, 0, array11->size*sizeof(unsigned int));
         DEBUG_NEW(array11->id);
      }
      array_t_param params0;
      params0.size = 1;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      params0.data[0] = array11;
      array_t* array12 = func41(&params0, loopsFactor);
      DEBUG_RETURN(array12->id);
      free(params0.data);
      for (int i = 0; i < array11->size; i++) {
         if (array11->data[i] == 73) { 
            return array11;
         }
      }
      array12->refC--;
      if(array12->refC == 0) {
         free(array12->data);
         free(array12);
         DEBUG_FREE(array12->id);
      }
      array11->refC--;
      if(array11->refC == 0) {
         free(array11->data);
         free(array11);
         DEBUG_FREE(array11->id);
      }
   }
   array_t* array14;
   if (pCounter > 0) {
      array14 = vars->data[--pCounter];
      array14->refC++;
      DEBUG_COPY(array14->id);
   } else {
      array14 = (array_t*)malloc(sizeof(array_t));
      array14->size = 862;
      array14->refC = 1;
      array14->id = 14;
      array14->data = (unsigned int*)malloc(array14->size*sizeof(unsigned int));
      memset(array14->data, 0, array14->size*sizeof(unsigned int));
      DEBUG_NEW(array14->id);
   }
   return array14;
}

File: data/multilang_comparison/program_nim/src/func29.nim
proc func29*(vars: ptr ArrayTParam, PATH0: uint64, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   let loopLimit5 = (50) div (4) + 1
   for loop5 in 0 ..< loopLimit5:
      discard
      if (PATH0 and 1'u64) != 0:
         discard
         var array8: ptr ArrayT
         if pCounter > 0:
            dec pCounter
            array8 = vars.data[pCounter]
            inc array8.refC
            debugCopy(array8.id)
         else:
            array8 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
            array8.size = 368
            array8.refC = 1
            array8.id = 8
            array8.data = cast[ptr UncheckedArray[uint32]](alloc(array8.size * sizeof(uint32)))
            zeroMem(array8.data, array8.size * sizeof(uint32))
            debugNew(array8.id)
         let loopLimit6 = (50) div (5) + 1
         for loop6 in 0 ..< loopLimit6:
            discard
            for i in 0 ..< array8.size:
               dec array8.data[i]
            for i in 0 ..< array8.size:
               if array8.data[i] == 82:
                  return array8
         var params0: ArrayTParam
         params0.size = 1
         params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
         params0.data[0] = array8
         var array9: ptr ArrayT = func46(addr params0, getPath(), loopsFactor)
         debugReturn(array9.id)
         dealloc(params0.data)
         dec array9.refC
         if array9.refC == 0:
            dealloc(array9.data)
            dealloc(array9)
            debugFree(array9.id)
         dec array8.refC
         if array8.refC == 0:
            dealloc(array8.data)
            dealloc(array8)
            debugFree(array8.id)
      else:
         discard
         var params0: ArrayTParam
         params0.size = 0
         params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
         var array37: ptr ArrayT = func40(addr params0, getPath(), loopsFactor)
         debugReturn(array37.id)
         dealloc(params0.data)
         dec array37.refC
         if array37.refC == 0:
            dealloc(array37.data)
            dealloc(array37)
            debugFree(array37.id)
      var array46: ptr ArrayT
      if pCounter > 0:
         dec pCounter
         array46 = vars.data[pCounter]
         inc array46.refC
         debugCopy(array46.id)
      else:
         array46 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
         array46.size = 567
         array46.refC = 1
         array46.id = 46
         array46.data = cast[ptr UncheckedArray[uint32]](alloc(array46.size * sizeof(uint32)))
         zeroMem(array46.data, array46.size * sizeof(uint32))
         debugNew(array46.id)
      var params0: ArrayTParam
      params0.size = 1
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      params0.data[0] = array46
      var array47: ptr ArrayT = func41(addr params0, loopsFactor)
      debugReturn(array47.id)
      dealloc(params0.data)
      if (PATH0 and 2'u64) != 0:
         discard
         var array50: ptr ArrayT
         if pCounter > 0:
            dec pCounter
            array50 = vars.data[pCounter]
            inc array50.refC
            debugCopy(array50.id)
         else:
            array50 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
            array50.size = 528
            array50.refC = 1
            array50.id = 50
            array50.data = cast[ptr UncheckedArray[uint32]](alloc(array50.size * sizeof(uint32)))
            zeroMem(array50.data, array50.size * sizeof(uint32))
            debugNew(array50.id)
         let loopLimit18 = (50) div (5) + 1
         for loop18 in 0 ..< loopLimit18:
            discard
            for i in 0 ..< array46.size:
               dec array46.data[i]
            for i in 0 ..< array50.size:
               if array50.data[i] == 29:
                  return array50
         var params1: ArrayTParam
         params1.size = 3
         params1.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params1.size * sizeof(ptr ArrayT)))
         params1.data[0] = array46
         params1.data[1] = array47
         params1.data[2] = array50
         var array51: ptr ArrayT = func54(addr params1, getPath(), loopsFactor)
         debugReturn(array51.id)
         dealloc(params1.data)
         dec array51.refC
         if array51.refC == 0:
            dealloc(array51.data)
            dealloc(array51)
            debugFree(array51.id)
         dec array50.refC
         if array50.refC == 0:
            dealloc(array50.data)
            dealloc(array50)
            debugFree(array50.id)
      else:
         discard
         var params1: ArrayTParam
         params1.size = 2
         params1.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params1.size * sizeof(ptr ArrayT)))
         params1.data[0] = array46
         params1.data[1] = array47
         var array61: ptr ArrayT = func50(addr params1, getPath(), loopsFactor)
         debugReturn(array61.id)
         dealloc(params1.data)
         dec array61.refC
         if array61.refC == 0:
            dealloc(array61.data)
            dealloc(array61)
            debugFree(array61.id)
      for i in 0 ..< array47.size:
         if array47.data[i] == 74:
            return array47
      dec array47.refC
      if array47.refC == 0:
         dealloc(array47.data)
         dealloc(array47)
         debugFree(array47.id)
      dec array46.refC
      if array46.refC == 0:
         dealloc(array46.data)
         dealloc(array46)
         debugFree(array46.id)
   var array67: ptr ArrayT
   if pCounter > 0:
      dec pCounter
      array67 = vars.data[pCounter]
      inc array67.refC
      debugCopy(array67.id)
   else:
      array67 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
      array67.size = 121
      array67.refC = 1
      array67.id = 67
      array67.data = cast[ptr UncheckedArray[uint32]](alloc(array67.size * sizeof(uint32)))
      zeroMem(array67.data, array67.size * sizeof(uint32))
      debugNew(array67.id)
   return array67

